lux/executor/PandasExecutor.py

```diff
@@ -163,17 +163,20 @@ def execute_aggregate(vis: Vis, isFiltered=True):
 
                 vis._vis_data = vis.data.reset_index()
                 # if color is specified, need to group by groupby_attr and color_attr
+
                 if has_color:
                     vis._vis_data = (
-                        vis.data.groupby([groupby_attr.attribute, color_attr.attribute], dropna=False)
+                        vis.data.groupby(
+                            [groupby_attr.attribute, color_attr.attribute], dropna=False, history=False
+                        )
                         .count()
                         .reset_index()
                         .rename(columns={index_name: "Record"})
                     )
                     vis._vis_data = vis.data[[groupby_attr.attribute, color_attr.attribute, "Record"]]
                 else:
                     vis._vis_data = (
-                        vis.data.groupby(groupby_attr.attribute, dropna=False)
+                        vis.data.groupby(groupby_attr.attribute, dropna=False, history=False)
                         .count()
```

Comment:
Since we're not recording in the history, does that mean that the history of the vis._data is no longer correct? I.e. if vis._data contains an aggregation, shouldn't its history say that it contains an aggregation?

 I think this brings up a bigger issue that the latest event should be recorded on the latest version of the dataframe, but I think we are recording the latest history on the dataframe that is transformed to get the latest dataframe. Is this correct?

Reply:
Not sure if I'm understanding correctly, but the history is tied to the user's DataFrame, not any DataFrames that we use internally like `vis.data`. The history is tied to both the DataFrame where the operation is acted on and also newly created DataFrames from that DataFrame. 

Now that I think about it, if I do something like

```python
df.groupby('Origin')
new_df = df.apply(f)
```
The `new_df` might have a groupby in its history, which wouldn't be correct. I'll look into this!

Reply:
I moved the history out of the original df, so it'll only exist in the groupby object and the DataFrame resulting from the groupby operation.